From 100dbd9a53efc0dd18a6e0f9dc7083ed584ffbee Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <rosslagerwall@gmail.com>
Date: Thu, 14 Nov 2013 16:50:46 +0200
Subject: [PATCH 05/11] sftp: Add "PermitLocalCommand no" to the OpenSSH
 commandline

Add "PermitLocalCommand no" to the OpenSSH commandline to prevent
user-configured commands running and interfering with the sftp session.
This is what the OpenSSH sftp client does.

https://bugzilla.gnome.org/show_bug.cgi?id=639599
---
 daemon/gvfsbackendsftp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/daemon/gvfsbackendsftp.c b/daemon/gvfsbackendsftp.c
index 99bfd5a..23d5b00 100644
--- a/daemon/gvfsbackendsftp.c
+++ b/daemon/gvfsbackendsftp.c
@@ -435,6 +435,7 @@ setup_ssh_commandline (GVfsBackend *backend)
     {
       args[last_arg++] = g_strdup ("-oForwardX11 no");
       args[last_arg++] = g_strdup ("-oForwardAgent no");
+      args[last_arg++] = g_strdup ("-oPermitLocalCommand no");
       args[last_arg++] = g_strdup ("-oClearAllForwardings yes");
       args[last_arg++] = g_strdup ("-oProtocol 2");
       args[last_arg++] = g_strdup ("-oNoHostAuthenticationForLocalhost yes");
-- 
1.9.0

